grails-app/controllers/au/org/ala/merit/ActivityController.groovy

```diff
@@ -239,7 +239,11 @@ class ActivityController {
         def activity = [activityId: "", siteId: siteId, projectId: projectId, progress: ActivityService.PROGRESS_PLANNED]
         def model = [activity: activity, returnTo: buildReturnToUrl(activity, params.returnTo, false), create: true]
         model.project = projectId ? projectService.get(projectId) : null
-        model.activityTypes = metadataService.activityTypesList(model.project?.associatedProgram, model.project?.associatedSubProgram)
+
+
+        Map program = (projectService.get(projectId)?.programId) ? programService.get(projectService.get(projectId)?.programId) : [:]
```

Comment:
This is very inefficient - we've just retrieved the project from ecodata on line 241, no need to query it again.